test/fail_compilation/retscope.d

```diff
@@ -267,3 +267,34 @@ void escape5() @safe
     pp = &p; // error
 }
 
+/***********************************************/
+
+/*
+TEST_OUTPUT:
+---
+fail_compilation/retscope.d(287): Error: escaping reference to local variable b
+---
+*/
+
+@safe int* foo6()(int* arg)
+{
+    return arg;
+}
+
+int* escape6() @safe
+{
+    int b;
+    return foo6(&b);
+}
+
+/***************************************************/
+
+struct S7
+{
+    int[10] a;
+    int[3] abc(int i)
```

Comment:
Would it change the design to extend scope checks (without scope inference) to unsafe code later on?
If not we could defer this decision and discussion.
Added it to the scope/DIP1000 backlog https://trello.com/c/XT4RECah/96-spec-how-to-extend-scope-checks-to-unsafe-code.